test_runner/src/main/kotlin/ftl/run/platform/RunAndroidTests.kt

```diff
@@ -38,9 +39,15 @@ internal suspend fun runAndroidTests(args: AndroidArgs): TestResult = coroutineS
 
     args.resolveApks().forEach { apks: ResolvedApks ->
         val testShards = apks.test?.let { test ->
-            AndroidTestShard.getTestShardChunks(args, test).also {
-                allTestShardChunks += it
+            AndroidTestShard.getTestShardChunks(args, test)
+        }
+        // We can't return if testShards is null since it can be a robo test.
+        if (testShards != null) {
+            if (!testShards.haveAtLeastOneTest) {
+                // No tests to run, skipping the execution.
+                return@forEach
             }
+            allTestShardChunks += testShards
```

Comment:
I am wondering if we could add only non-empty shards here, let me know what you think

Reply:
Actually, it will add only non-empty shards since otherwise `testShards.haveAtLeastOneTest` would return false and `return@forEach` will return before adding those shards

Reply:
`haveAtLeastOneTest` uses `any {...}` which returns true if at least one item matches predicate

Reply:
I've got you, my bad, I didn't think about that case. I'm refactoring it thanks!